File: frontend/src/app/services/rendez-vous.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class RendezVousService {

  constructor(private http: HttpClient) { }

  getPatientByIdRendezVous(idRendezVous:number): Observable<any> {
    return this.http.get('http://localhost:8085/api/admin/mesRendezVous/getPatientByIdRendezVous'+"/"+idRendezVous)
  }
  getListRendezVousByIdPatient(idPatient:number): Observable<any> {
    return this.http.get('http://localhost:8085/api/admin/mesRendezVous/listRendezVousByIdPatient'+"/"+idPatient)
  }
  getRendezVousAnnulerByIdPatient(idPatient:number): Observable<any> {
    return this.http.get('http://localhost:8085/api/admin/mesRendezVous/listRendezVousAnnulerByIdPatient'+"/"+idPatient)
  }
  getRendezVousRealiserByIdPatient(idPatient:number): Observable<any> {
    return this.http.get('http://localhost:8085/api/admin/mesRendezVous/listRendezVousRealiserByIdPatient'+"/"+idPatient)
  }
  getRendezVousEnAttenteByIdPatient(idPatient:number): Observable<any> {
    return this.http.get('http://localhost:8085/api/admin/mesRendezVous/listRendezVousEnAttenteByIdPatient'+"/"+idPatient)
  }
  listRendezVousByDateAndIdPatient(idPatient:number,dateRendezVous:string): Observable<any> {
    let queryParams = new HttpParams();
    queryParams = queryParams.append('dateRendezVous', dateRendezVous);
    return this.http.get('http://localhost:8085/api/admin/mesRendezVous/listRendezVousByDateAndIdPatient'+"/"+idPatient,{ params: queryParams })
  }
  saveRendezVous(rendezVous:any,idPatient:number): Observable<any>{
    return this.http.post('http://localhost:8085/api/admin/mesRendezVous/newRendezVous'+"/"+idPatient,rendezVous)
  }
  getAllRendezVous(): Observable<any> {
    return this.http.get('http://localhost:8085/api/admin/mesRendezVous/listRendezVous')
  }
  getRendezVousAnnuler(): Observable<any> {
    return this.http.get('http://localhost:8085/api/admin/mesRendezVous/listRendezVousAnnuler')
  }
  getRendezVousRealiser(): Observable<any> {
    return this.http.get('http://localhost:8085/api/admin/mesRendezVous/listRendezVousRealiser')
  }
  getRendezVousEnAttente(): Observable<any> {
    return this.http.get('http://localhost:8085/api/admin/mesRendezVous/listRendezVousEnAttente')
  }
  getRendezVousByNomPatient(nom:string): Observable<any> {
    let queryParams = new HttpParams();
    queryParams = queryParams.append('nom', nom);
    return this.http.get('http://localhost:8085/api/admin/mesRendezVous/listRendezVousByNomPatient',{ params: queryParams })
  }
  getRendezVousNonAnnuler(): Observable<any> {
    return this.http.get('http://localhost:8085/api/admin/mesRendezVous/listRendezVousNonAnnuler')
  }
  annulerRendezVous(idRendezVous:number,newRendezVous:any): Observable<any> {
    return this.http.put('http://localhost:8085/api/admin/mesRendezVous/annulerRendezVous'+'/'+idRendezVous,newRendezVous)
  }
  updateRendezVous(idRendezVous:number,newRendezVous:any): Observable<any> {
    return this.http.put('http://localhost:8085/api/admin/mesRendezVous/updateRendezVous'+'/'+idRendezVous,newRendezVous)
  }
  getDisponibilteRendezVous(dateRendezVous:string, heureDebut:string): Observable<any> {
    let queryParams = new HttpParams();
    queryParams = queryParams.append('heureDebut', heureDebut);
    queryParams = queryParams.append('dateRendezVous', dateRendezVous);
    return this.http.get('http://localhost:8085/api/admin/mesRendezVous/disponibiliterRendezVous',{ params: queryParams })
  }
}

File: gestionCabinet/src/main/java/com/example/demo/services/RendezVousService.java
package com.example.demo.services;

import java.util.Date;
import java.util.List;

import com.example.demo.dtos.PatientDTO;
import com.example.demo.dtos.RendezVousDTO;

public interface RendezVousService {
	
	RendezVousDTO saveRendezVous(RendezVousDTO rendezVou,Long idPatient);
	RendezVousDTO modifierRendezVous(RendezVousDTO rendezVousDTO,Long idPatient);
	RendezVousDTO annulerRendezVous(RendezVousDTO rendezVousDTO,Long idPatient);
	List<RendezVousDTO> listRendezVousEnAttente();
	List<RendezVousDTO> listRendezVousRealiser();
	List<RendezVousDTO> listRendezVousAnnuler();
	PatientDTO getPatientByIdRendezVous(Long idRendezVous);
	List<RendezVousDTO> listRendezVous();
	List<RendezVousDTO> listRendezVousByNomPatient(String nom);
	List<RendezVousDTO> listRendezVousNonAnnuler();
	List<RendezVousDTO> listRendezVousEnAttenteByNomPatient(String nom);
	RendezVousDTO disponibilite(Date dateRendezVous, String heureDebut);
	List<RendezVousDTO> listRendezVousByIdPatient(Long idPatient);
	List<RendezVousDTO> getRendezVousByDateAndIdPatient(String date,Long idPatient) throws Exception;
	List<RendezVousDTO> listRendezVousEnAttenteByIdPatient(Long idPatient);
	List<RendezVousDTO> listRendezVousRealiserByIdPatient(Long idPatient);
	List<RendezVousDTO> listRendezVousAnnulerByIdPatient(Long idPatient);
	void rappelRendezVous() throws Exception;
	void email(String toEmail, String subject, String body);
}
